subprojects/resources-s3/src/test/groovy/org/gradle/internal/resource/transport/aws/s3/S3ClientTest.groovy

```diff
@@ -98,8 +98,17 @@ class S3ClientTest extends Specification {
         objectListing.getCommonPrefixes() >> ['/SNAPSHOT/', '/SNAPSHOT/1.0.8/']
 
         when:
-        def results = s3Client.resolveResourceNames(objectListing)
+        def results = s3Client.resolveDirectoryResourceNames(objectListing)
```

Comment:
The test exercises private methods. Please move the methods resolveResourceNames, resolveDirectoryResourceNames and resolveFileResourceNames to a new object (e.g. S3ResourceResolver), have them public there and then have the Test test the new class.